### PR TITLE
UNIX socket support, updated README, added inline documentation for :uri

### PR DESCRIPTION
**Pull request by cognitiveflux**

### CODE REVIEW

**Comment by cognitiveflux** [comment]
I didn't test with Autobahn but I did test local client-server connections using `websocket-eventmachine-client` and `websocket-eventmachine-server` for both TCP and UNIX sockets.

**Comment by imanel** [comment]
Wouldn't it be clearer if `socket` option would be added instead of modifying meaning of `host`? It could disable other checks (host/port/ssl etc) and directly call [connect_unix_domain](http://www.rubydoc.info/github/eventmachine/eventmachine/EventMachine.connect_unix_domain) (I know it's alias, but it would be much clearer).

**Comment by cognitiveflux** [comment]
I agree, I originally wrote it accordingly:

```ruby
 def self.connect(args = {})
  if args.key?(:socket)
   ::EventMachine.connect_unix_domain args[:socket], nil, self, args
  else
   # Previous ::EventMachine.connect logic
  end
 end
```

However when I ran the _Hello Server_ client example, it would never complete the callbacks **unless** a `:host` attribute was set:

**Doesn't work:**
```ruby
EM.run do
 ws = WebSocket::EventMachine::Client.connect(:socket => '/tmp/wssock')
 ws.onopen do
   puts "Connected"
 end

 ws.onmessage do |msg, type|
   puts "Received message: #{msg}"
 end

 ws.onclose do |code, reason|
   puts "Disconnected with status code: #{code}"
 end

 EventMachine.next_tick do
   ws.send "Hello Server!"
 end

end
```

Output (just waits):
```bash
> ./client.rb
```

**Works:**
```ruby
EM.run do
 ws = WebSocket::EventMachine::Client.connect(:socket => '/tmp/wssock', :host => 'foo')
# ...
```

Output:
```bash
> ./client.rb
Connected
Received message: Hello Server!
```

I'm not sure what in the code base has a strict dependency on the `:host` attribute.

Even if I change the [original call](https://github.com/imanel/websocket-eventmachine-client/blob/master/lib/websocket/eventmachine/client.rb#L42) to explicitly use a socket path (not passed in from `args`, as shown below) the client still hangs unless the `:host` option is present:

```ruby
::EventMachine.connect '/path/to/socket', nil, self, args
```

With the above change, `WebSocket::EventMachine::Client.connect()` fails but `WebSocket::EventMachine::Client.connect(host: 'foo')` succeeds.

Clearly the `:host` option is not being used because a value of 'foo' is invalid, but the `:host` option must be present.

Thoughts?

**Comment by imanel** [comment]
This is because of WebScoket specification - you always need host provided, otherwise request will be dropped. I'm not sure how other implementations are handling sockets, but I'm pretty sure that could be categorized as uncommon usage ;) I would suggest adding setting some default host if not provided (`localhost`?)

**Comment by cognitiveflux** [comment]
I created another method `connect_unix_domain`, following the design of `EventMachine`, that is intended for socket-only connections and leaves the normal `connect` method untouched.

File: lib/websocket/eventmachine/client.rb [hunk 1]
```diff
@@ -41,6 +42,18 @@ def self.connect(args = {})
...
+      
+      # Make a websocket connection to a UNIX-domain socket.
+      # @param socketname [String] Unix domain socket (local fully-qualified path) you want to use
+      # @param args [Hash] Arguments for connection      
+      # @option args [Integer] :version Version of protocol to use(default = 13)
+      # @option args [Hash] :headers HTTP headers to use in the handshake
+      # @option args [Boolean] :ssl Force SSL/TLS connection
+      def self.connect_unix_domain(socketname, args = {})
+        fail ArgumentError, 'invalid socket' unless File.socket?(socketname)
+        args[:host] = 'localhost'
```
Comment (imanel) [inline lib/websocket/eventmachine/client.rb]: Should probably be `||=`

**Comment by imanel** [comment]
One small change and I'll merge & release it. Thanks a lot for your work!

**Comment by imanel** [comment]
Released as 1.2.0. Thanks for your work!

**Pull request merged**